src/workloads/docs/CrudActorEncrypted.yml

```diff
@@ -0,0 +1,166 @@
+SchemaVersion: 2018-07-01
+Owner: "@10gen/server-security"
+Description: |
+  Example workload demonstrating encryption support in Genny's client pools.
+  The CrudActor in this workload performs encrypted operations using a Client
+  that is configured with the FLE and Queryable auto encryption options.
+
+  Encryption is configured per Client. Under the EncryptionOptions
+  element of a Client instance, the following fields must be provided:
+
+    KeyVaultDatabase     - the name of the keyvault database
+    KeyVaultCollection   - the name of the keyvault collection
+    EncryptedCollections - sequence of encrypted collection descriptors
+
+  Each encrypted collection descriptor under the EncryptedCollections sequence
+  must define the following fields:
+
+    Database           - the name of the encrypted database
+    Collection         - the name of the encrypted collection
+    EncryptionType     - the type of client-side encryption used (either 'fle' or 'queryable')
+
+  Depending on the EncryptionType, the encryption schema may be specified using either:
+    FLEEncryptedFields        - if EncryptionType is "fle"
+    QueryableEncryptedFields  - if EncryptionType is "queryable"
+
+  These map the path of an encrypted field in the collection to its encryption parameters.
+  Both can appear under a collection descriptor node, but only the node that matches the specified
+  EncryptionType will apply throughout the workload.
+
+  Each subnode of FLEEncryptedFields or QueryableEncryptedFields is a key-value pair where
+  the key is the field path, and the value is an object whose required fields depend on the type.
+
+  Field paths cannot be a prefix of another path under the same FLEEncryptedFields or
+  QueryableEncryptedFields map. For instance, "pii" and "pii.ssn" are not allowed, since "pii" is
+  a path prefix of the other.
+
+  Under FLEEncryptedFields, each object mapped to a path must have the following fields:
+    type      - the BSON type of values for this field
+    algorithm - the encryption algorithm: either "random" or "deterministic"
+                "deterministic" means that the ciphertext is always the same for the same
+                plaintext value, which means that this field can be used on queries.
+
+  Under QueryableEncryptedFields, each object mapped to a path must have the following fields:
+    type      - the BSON type of values for this field
+    queries   - an array of objects that specify "queryType" and other parameters for QE.
+
+  An Actor that wishes to perform encrypted operations on an encrypted collection must
+  be using the Client pool under which the schema for that encrypted collection is defined.
```

Comment:
The way this is currently structured does not allow for 2 pools to share the same schema, and if two pools on the same URI happen to have the same copy of an encrypted collection's schema, then they would each generate their own data keys, which is not good.

Reply:
I think to accomplish sharing between two pools, we'd need to have a global `EncryptionManager` that will handle all the global initialization (key vault & data key creation). Afterwards, pools can reference any schema owned by the EncryptionManager in order to use encrypted operations.

Reply:
Gotcha. Yeah whether that's worth it or not I couldn't say, but maybe someone on product perf would have an idea of whether that'd constrain the tests they run by too much?

If that doesn't get implemented, I'd recommend pointing out explicitly in the docs that it's not supported.

Reply:
I implemented EncryptionManager anyways since we'll probably going run into this exact situation you described when we're writing our QE workloads. Plus, this meant that I no longer need to touch the Pools maps in PoolManager.